ssl-config/src/test/java/com/palantir/ssl/SslSocketFactoriesConnectionTests.java

```diff
@@ -0,0 +1,482 @@
+    public void testSslWithClientAuthenticationJks() {
+        TrustStoreConfiguration serverTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration serverKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.SERVER_KEY_STORE_JKS_PATH)
+                .password(TestConstants.SERVER_KEY_STORE_JKS_PASSWORD)
+                .build();
+
+        SslConfiguration serverConfig = SslConfiguration.builder()
+                .trust(serverTrustStoreConfig)
+                .key(serverKeyStoreConfig)
+                .build();
+
+        TrustStoreConfiguration clientTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration clientKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.CLIENT_KEY_STORE_JKS_PATH)
+                .password(TestConstants.CLIENT_KEY_STORE_JKS_PASSWORD)
+                .build();
+
+        SslConfiguration clientConfig = SslConfiguration.builder()
+                .trust(clientTrustStoreConfig)
+                .key(clientKeyStoreConfig)
+                .build();
+
+        runSslConnectionTest(serverConfig, clientConfig, ClientAuth.WITH_CLIENT_AUTH);
+    }
+
+    @Test
+    public void testSslWithClientAuthenticationPkcs12() {
+        TrustStoreConfiguration serverTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration serverKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.SERVER_KEY_STORE_P12_PATH)
+                .type(TestConstants.SERVER_KEY_STORE_P12_TYPE)
+                .password(TestConstants.SERVER_KEY_STORE_P12_PASSWORD)
+                .build();
+
+        SslConfiguration serverConfig = SslConfiguration.builder()
+                .trust(serverTrustStoreConfig)
+                .key(serverKeyStoreConfig)
+                .build();
+
+        TrustStoreConfiguration clientTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration clientKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.CLIENT_KEY_STORE_P12_PATH)
+                .type(TestConstants.CLIENT_KEY_STORE_P12_TYPE)
+                .password(TestConstants.CLIENT_KEY_STORE_P12_PASSWORD)
+                .build();
+
+        SslConfiguration clientConfig = SslConfiguration.builder()
+                .trust(clientTrustStoreConfig)
+                .key(clientKeyStoreConfig)
+                .build();
+
+        runSslConnectionTest(serverConfig, clientConfig, ClientAuth.WITH_CLIENT_AUTH);
+    }
+
+    @Test
+    public void testSslWithClientAuthenticationMixed() {
```

Comment:
would love to also see this test against a dropwizard/feign stack (since that is really the default use-case for this library). have a look at the other sub-projects for a simple test setup.